### PR TITLE
Rd chart sect one popups 

### PR DESCRIPTION
**Pull request by xjackk**
* https://federal-spending-transparency.atlassian.net/browse/DA-4463
* https://federal-spending-transparency.atlassian.net/browse/DA-4464
* https://federal-spending-transparency.atlassian.net/browse/DA-5626

Addressing all the tickets for rd section one!

### CODE REVIEW

File: src/page-sections/rd-and-contracting/spending/spendingChart.jsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    });
+  };
+
+  /* 
+    - Dynamic Event Listener - 
+    Use "e.target" and not element directly as some are not drawn
+    on the DOM yet. 
+  */
+  detailsListener = (e) => {
+    let element = e.target;
```
Comment (brett-varney) [inline src/page-sections/rd-and-contracting/spending/spendingChart.jsx:59]: Could e.target ever be null? Just preventing possible console errors

File: src/page-sections/rd-and-contracting/spending/spendingChart.jsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      that.setState({ showDetails: false });
+    });
+  };
+
+  /* 
+    - Dynamic Event Listener - 
+    Use "e.target" and not element directly as some are not drawn
+    on the DOM yet. 
+  */
+  detailsListener = (e) => {
```
Comment (brett-varney) [inline src/page-sections/rd-and-contracting/spending/spendingChart.jsx:58]: This event isn't firing when space or enter is hit on the SHOW DETAILS; enter does nothing and space scrolls down.

**Comment by brett-varney** [comment]
In general, very nice, even IE!. Specific issues I've found:

- SHOW DETAILS is now keyboard-focusable, but not keyboard-activated. Enter does nothing, space scrolls down

- pop-up is not activated clicking inside the box, just the border, person & SHOW DETAILS

- the dashed chart only highlights when hovering on the border, not inside the box

- pop-up close isn't keyboard-accessible, should close when focused and enter, space or esc hit

- pop-up close only works clicking exactly on the X, not in the whitespace inside the X (it doesn't have a surrounding block that can be clicked)

- tablet ONLY: same issue as above with SHOW DETAILS words

- on all my non-Chrome browsers, the highlight is opaque:
image

File: src/page-sections/rd-and-contracting/spending/spendingChart.jsx (hunk 1 above)
Comment (xjackk) [reply src/page-sections/rd-and-contracting/spending/spendingChart.jsx:59]: yep, could do a check here, good call

File: src/page-sections/rd-and-contracting/spending/spendingChart.jsx (hunk 2 above)
Comment (xjackk) [reply src/page-sections/rd-and-contracting/spending/spendingChart.jsx:58]: haven't added keyboard stuff yet, forgot, good catch. Will wire up enter and etc for keyboard navigation

Comment (brett-varney) [reply src/page-sections/rd-and-contracting/spending/spendingChart.jsx]: I think keyboard works acceptably now. It opens & closes with Enter, not with Space or Esc. Those might be issues, but minor in comparison.

**Comment by brett-varney** [comment]
Current issues:

- pop-up activates clicking inside the box only if the box is filled, which only happens when hovering on a solid segment of the border (icon & text still work fine)

- box highlight is solid & opaque on all browsers except Chrome (Win 10)

- pop-up close still only registers click when right on the X lines; hard to hit

- TABLET ONLY: pop-up still activates only when clicking on paths of the words

**Comment by brett-varney** [comment]
Oh, and merge conflicts. @anjenkin might be dealing with these in her chart updates...

**Comment by brett-varney** [comment]
> 
> 
> Current issues:
> 
>     * pop-up activates clicking inside the box only if the box is filled, which only happens when hovering on a solid segment of the border (icon & text still work fine)
> 
>     * box highlight is solid & opaque on all browsers except Chrome (Win 10)
> 
>     * pop-up close still only registers click when right on the X lines; hard to hit
> 
>     * TABLET ONLY: pop-up still activates only when clicking on paths of the words

I know this is a pain, and I'm sorry you got stuck with it. I reconciled all the other updates, merged latest dev, fixed merge conflicts, so start by pulling this branch, or you'll have to redo all that.

In my local testing, the above still happen except the region is highlighted and clickable when hovering, but only in Chrome, so it may be a non-Webkit thing. And now:

- the box is red when NOT hovered on (except in Chrome)

- the words disappear when hovered on, tablet only, all browsers

**Pull request merged**